### PR TITLE
Plugin environment variables

### LINKED ISSUE #632: Inform plugin authors about available environment variables
We should update the [plugin authoring docs](https://developer.fermyon.com/spin/plugin-authoring#authoring-a-spin-plugin) to inform that the following environment variables (with example values) are set by spin in the plugin child process:
```sh
SPIN_BIN_PATH=/Users/cardoso/.cargo/bin/spin
SPIN_BRANCH=feat/more_build_info
SPIN_BUILD_DATE=2023-05-15
SPIN_COMMIT_DATE=2023-05-15
SPIN_COMMIT_SHA=49fb11b
SPIN_DEBUG=false
SPIN_TARGET_TRIPLE=aarch64-apple-darwin
SPIN_VERSION=1.2.0-pre0
SPIN_VERSION_MAJOR=1
SPIN_VERSION_MINOR=2
SPIN_VERSION_PRE=pre0
```

### PR DESCRIPTION
**Pull request by itowlson**
Fixes #632.

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [ ] The `title, `template`, and `date` are all set
- [ ] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [ ] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [ ] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [ ] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [ ] Headings are using Title Case
- [ ] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors
- [ ] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

**Review by kate-goldenring — APPROVED** [review]

File: content/spin/plugin-authoring.md [hunk 1]
```diff
@@ -70,6 +71,24 @@ Spin plugins are implemented as a manifest that points to one or more `.tar.gz`
 1. Create tar archives of the executables for the platforms you want to support
 2. Compose a manifest that describes the plugin and lists the URLs for those tar archives
 
+### Environment Variables Available to the Plugin Executable
+
+When Spin launches a plugin executable, it sets a number of environment variables on the plugin process. These contain information about the Spin binary that the user ran to launch your plugin. Your plugin can use them to detect the Spin environment and adjust its behavior if necessary. The variables set are:
```
Comment (tpmccallum) [inline content/spin/plugin-authoring.md:76]: Regarding the `that the user ran to launch your plugin` are we talking about environment variables that have already been created based on another user's machine i.e. when I launch, my system's environment variables get set behind the scenes (to whatever the creator of the plugin pre-programmed)?

Comment (itowlson) [reply content/spin/plugin-authoring.md:76]: Let me explain the context and maybe you can help me improve the wording...

* Suppose there is a plugin called `honk`.
* When the user runs `spin honk`, that runs `spin`, passing the subcommand `honk`.
* Spin recognises that `honk` is not a built-in command, so it checks if it has a plugin called `honk`. Which it does.
* Spin starts the `honk` plugin as a child process.  As part of starting this process, Spin sets a bunch of environment variables on the child process.  These EVs contain information about the current Spin process (the parent).
* The `honk` plugin code runs, with these environment variables set.

The `spin` process in the second bullet, i.e. the one that the user ran when they typed `spin honk`, is what I meant by "the Spin binary that the user ran to launch your plugin."

The EVs come from _this_ Spin process, not from anything programmed by the plugin author.  It's like, as Spin kicks off the plugin, it tells the plugin "hey did you know you can call me back at /home/ivan/github/spin/target/debug/spin? and that my version number is 1.4.0?"

Does that make more sense... or _even less sense_...?

File: content/spin/plugin-authoring.md [hunk 2]
```diff
@@ -70,6 +71,30 @@ Spin plugins are implemented as a manifest that points to one or more `.tar.gz`
 1. Create tar archives of the executables for the platforms you want to support
 2. Compose a manifest that describes the plugin and lists the URLs for those tar archives
 
+### Environment Variables Available to the Plugin Executable
+
+Your plugin may need to know information about the instance of Spin it's running in. For example, suppose your plugin wants to call `spin build`. The trouble is that you don't know if it's on the user's system PATH. Suppose, further, than your plugin would prefer to call `spin build -c` (to build only a specific component) if it's available, but can fall back to `spin build` (to build everything) if it's not. The `-c` option only exists in Spin 1.4 and above, so this optimization requires that you know which version of Spin you're running in.
```
Comment (tpmccallum) [inline content/spin/plugin-authoring.md:76]:
```suggestion
Your plugin may need to know information about the instance of Spin it's running in. For example, suppose your plugin wants to call `spin build`. The trouble is that you don't know if it's on the user's system PATH. Suppose, further, that your plugin would prefer to call `spin build -c` (to build only a specific component) if it's available but can fall back to `spin build` (to build everything) if it's not. The `-c` option only exists in Spin 1.4 and above, so this optimization requires that you know which version of Spin you're running in.
```

**Review by tpmccallum — COMMENTED** [review]
Just a single word fix; and otherwise LGTM.

**Review by tpmccallum — APPROVED** [review]

**Pull request merged**